archinstall/lib/disk/partitioning_menu.py

```diff
@@ -288,8 +289,8 @@ def handle_action(
 					partition.mountpoint = self._prompt_mountpoint()
 					if partition.mountpoint == Path('/boot'):
 						partition.set_flag(PartitionFlag.BOOT)
-						if self._using_gpt:
-							partition.set_flag(PartitionFlag.ESP)
+					if self._using_gpt and partition.type_uuid == PartitionGUID.ESP.bytes:
+						partition.set_flag(PartitionFlag.ESP)
```

Comment:
Why add a check that the partition is the type that it is being set to? How would that make sense?

Reply:
i believe those flags are internal flags, no?

Reply:
This is code for the setting a partition as an ESP. It could be a partition that has yet to be created and so does not exist.